### PR TITLE
[FIX] Fix EnumValidator

### LINKED ISSUE #78: [BUG] Possible problem with the enum validator
If I add the following test, a test for a direct specification of a cluster method, I get an error instead of the expected commented out stuff?
It looks like there is something wrong with our new validator.

```cpp
TEST_F(detect_breakends, with_simple_clustering_method)
{
    cli_test_result result = execute_app("detect_breakends",
                                         data("simulated.minimap2.hg19.coordsorted_cutoff.sam"),
                                         "detect_breakends_insertion_file_out.fasta",
                                         "-c 0");
    std::string expected_err
    {
        "[Error] Validation failed for option -c/--clustering_method: Value simple_clustering is not one of [].\n"
        // "INS1: Reference\tchr21\t41972616\tForward\tRead\t0\t2294\tForward\tm2257/8161/CCS\n"
        // "INS2: Reference\tchr21\t41972616\tReverse\tRead\t0\t3975\tReverse\tm2257/8161/CCS\n"
        // "BND: Reference\tchr22\t17458417\tForward\tReference\tchr21\t41972615\tForward\tm41327/11677/CCS\n"
        // "BND: Reference\tchr22\t17458418\tForward\tReference\tchr21\t41972616\tForward\tm21263/13017/CCS\n"
        // "BND: Reference\tchr22\t17458418\tForward\tReference\tchr21\t41972616\tForward\tm38637/7161/CCS\n"
        // "Start clustering...\n"
        // "Done with clustering. Found 4 junction clusters.\n"
        // "No refinement was selected.\n"
    };
    EXPECT_NE(result.exit_code, 0);
    // EXPECT_EQ(result.exit_code, 0);
    EXPECT_EQ(result.out, "");
    // EXPECT_EQ(result.out, expected_res);
    EXPECT_EQ(result.err, expected_err);
}
```

### PR DESCRIPTION
**Pull request by Irallia**
Resolves #78 

Update EnumValidator such that `args.methods` (detection_methods) can be used and give a nice help page view.

EDIT (22.03.2021): We should probably wait for https://github.com/seqan/seqan3/issues/2464 and update the seqan3 library.
EDIT (06.04.2021): We also want to wait for https://github.com/seqan/seqan3/pull/2502 to avoid another seqan3 version change.
EDIT (07.04.2021): Both seqan PRs are merged now. This PR can get merged after the reviews.

### CODE REVIEW

**Review by eseiler — APPROVED** [review]

File: test/cli/iGenVar_cli_test.cpp [hunk 1]
```diff
@@ -224,6 +268,33 @@ TEST_F(detect_breakends, with_detection_method_duplicate_arguments)
...
+TEST_F(detect_breakends, test_unknown_argument)
+{
+    cli_test_result result = execute_app("iGenVar",
+                                         "-j", data(default_alignment_long_reads_file_path),
+                                         fasta_out_file_path,
+                                         "-m 9");
+    std::string expected_err
+    {
+        "[Error] Value parse failed for -m: Argument 9 could not be parsed as type std::string.\n"
+    };
```
Comment (Irallia) [inline test/cli/iGenVar_cli_test.cpp:296]: This unhelpful error message is fixed by issue https://github.com/seqan/seqan3/issues/2464.

File: test/cli/iGenVar_cli_test.cpp [hunk 2]
```diff
@@ -103,54 +100,107 @@ std::string expected_res
...
+    "Start clustering...\n"
+    "Done with clustering. Found 3 junction clusters.\n"
+    "No refinement was selected.\n"
+};
+
+TEST_F(iGenVar_cli_test, no_options)
 {
     cli_test_result result = execute_app("iGenVar");
-    std::string expected
+    std::string expected_res
```
Comment (eldariont) [inline test/cli/iGenVar_cli_test.cpp:117]: This local variable has the same name as the global variable above. Maybe rename to `expected_res_no_options`?

Comment (eldariont) [reply test/cli/iGenVar_cli_test.cpp:117]: Or we move the global variable into the test cases which use it? I guess we could decide to either define all result and error messages above the test cases or inside each test case. A mix of both is confusing.

**Review by eldariont — CHANGES_REQUESTED** [review]
Looks very good. I only have one tiny comment about the test cases.

**Review by eldariont — APPROVED** [review]
Thanks!

**Pull request merged**